Makefile

```diff
@@ -2475,6 +2475,19 @@ LIB_PERL_GEN := $(patsubst perl/%.pm,perl/build/lib/%.pm,$(LIB_PERL))
 LIB_CPAN := $(wildcard perl/FromCPAN/*.pm perl/FromCPAN/*/*.pm)
 LIB_CPAN_GEN := $(patsubst perl/%.pm,perl/build/lib/%.pm,$(LIB_CPAN))
 
+# Set MSWIN32_PERL_PATH=false to disable search
+ifndef MSWIN32_PERL_PATH
+MSWIN32_PERL_PATH = $(shell \
+	for perl in $$(type -ap perl); do \
+		if "$$perl" -e 'exit 1 if $$^O ne q{MSWin32}'; then \
+			echo "$$perl"; \
+			exit 0; \
+		fi; \
+	done \
+)
+endif
+MSWIN32_PERL_PATH_SQ = $(subst ','\'',$(MSWIN32_PERL_PATH))
```

Comment:
I am not sure that this is a good change. By default, the Git for Windows SDK comes with a `perl.exe` and we know exactly which one it is: the one that is shipped with (and therefore, supported by) Git for Windows.

In other words, if you really want this, I would want to make it an opt-in rather than an opt-out.

Besides, would it not make things substantially cleaner if you simply prepended the path to the known MSWin32 Perl (if configured via `MSWIN32_PERL_PATH` explicitly) to `PATH`? That way, `perl` would already find the Perl interpreter you want, no need to play extra games with any `mswin32_perl` shell function.